plugins/processors/starlark/starlark_test.go

```diff
@@ -3387,3 +3308,29 @@ func testLoadFunc(module string, logger telegraf.Logger) (starlark.StringDict, e
 func testNow(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
 	return starlarktime.Time(time.Date(2021, 4, 15, 12, 0, 0, 999, time.UTC)), nil
 }
+
+func newStarlarkFromSrouce(source string) *Starlark {
+	inner := common.NewStarlarkCommon(testLoadFunc)
+	inner.Log = testutil.Logger{}
+	inner.Source = source
+	return &Starlark{
+		StarlarkCommon: inner,
+	}
+}
```

Comment:
Typo here, furthermore I would initialize the inner struct directly using
```suggestion
func newStarlarkFromSource(source string) *Starlark {
	plugin := Starlark {
		StarlarkCommon: common.StarlarkCommon{
			starlarkLoadFunc: testLoadFunc,
			Log: testutil.Logger{},
			Source: source,
		},
	}
	return &plugin
}
```
You could then avoid setting the source and this way merge `newStarlarkFromSrouce`, `newStarlarkFromScript` and `newStarlarkNoScript` by just setting the corresponding field in the caller. Or you move this code to the caller directly.

Reply:
If so `starlarkLoadFunc` must be public